diff --git a/example/api2-samples/api2-decode-encode-audio.cpp b/example/api2-samples/api2-decode-encode-audio.cpp
index 04ff12e..6063888 100644
--- a/example/api2-samples/api2-decode-encode-audio.cpp
+++ b/example/api2-samples/api2-decode-encode-audio.cpp
@@ -190,7 +190,7 @@ int main(int argc, char **argv)
                 break;
             }
 
-            if (pkt.streamIndex() != audioStream) {
+            if (pkt && pkt.streamIndex() != audioStream) {
                 continue;
             }
 
